async_simple/Common.h

```diff
@@ -31,9 +36,14 @@ namespace async_simple {
 // a bug in the library. If logicAssert fails, it means
 // there is a bug in the user code.
 inline void logicAssert(bool x, const char* errorMsg) {
-    if (FL_LIKELY(x)) {
+    if (x)
+#if defined(__clang__) && __clang_major__ < 12
+    {
         return;
     }
+#else
+        [[likely]] { return; }
+#endif
```

Comment:
This looks redundant. I would suggest:
```suggestion
    if (x) AS_LIKELY
        return;
```

Reply:
Good